File: java/ToolGood/Words/IllegalWordsSearchResult.java
package ToolGood.Words;


public class IllegalWordsSearchResult
{
    public IllegalWordsSearchResult(String keyword, int start, int end, String srcText)
    {
        Keyword = keyword;
        Success = true;
        End = end;
        Start = start;
        SrcString = srcText.substring(Start, end+1);
    }
    public IllegalWordsSearchResult(String keyword, int start, int end, String srcText, int type)
    {
        Keyword = keyword;
        Success = true;
        End = end;
        Start = start;
        SrcString = srcText.substring(Start, end+1);
        BlacklistType = type;
    }

    public IllegalWordsSearchResult()
    {
        Success = false;
        Start = 0;
        End = 0;
        SrcString = null;
        Keyword = null;
    }
    public Boolean Success ;
    public int Start;
    public int End ;
    public String SrcString ;
    public String Keyword;
    public int BlacklistType ;
}

File: golang/ToolGood/Words/IllegalWordsSearchResult.go
package Words

type IllegalWordsSearchResult struct {
	Start int
	End int
	Keyword string
	BlacklistType int
	SrcString string
}
 
func NewIllegalWordsSearchResult(keyword string,start int,end int,srcText string,_type int) *IllegalWordsSearchResult  {
	return &IllegalWordsSearchResult{
		Start:start,
		End:end,
		Keyword:keyword,
		SrcString:string ( []rune(srcText)[start:end]),
		BlacklistType:_type,
	}
}
 